src/blocks/battery.rs

```diff
@@ -120,9 +119,55 @@ impl Block for Battery {
 
         let state = read_file(&format!("{}status", self.device_path))?;
 
+
+        #[allow(unused_assignments)]
+        let mut energy_now = 0;
```

Comment:
I'm not sure this annotation is necessary. This code could probably be refactored to be more idiomatic and remove the need for this annotation, e.g.
```rust
let energy_now = if file_exists ... { ... } else { 0 };
```
The same is true for the other let bindings here, unless I'm very much mistaken. You can use tuple destructuring (e.g. `let (hours, minutes) = if ... { } else { }`) as well to fix the one below.

Reply:
The ````(hours,minutes)```` thing is a bit more tricky because I reuse ````hours```` when calculating minutes.